### PR TITLE
Update everything

### PR DESCRIPTION
**Pull request by ChuckWoodraska**
- Remove outdated code for versions of Python that are no longer supported
- Add in newer versions of Python for testing
- Update dependencies
- Rerun all the VCR tests for new test data
- Change a few snippets of code to follow best practices
- Fix spacing that the linter was complaining about
- Update test harness to handle how updated dependencies work

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -410,7 +395,7 @@ def test_risk():
 def test_risk_evidence():
     with api.risk_evidence(domain='google.com') as result:
         assert result
-        assert list(result) == [{'name': 'whitelist', 'risk_score': 0}]
+        assert list(result) == [{'name': 'zerolist', 'risk_score': 0}]
```
Comment (ChuckWoodraska) [inline tests/test_api.py:398]: Updated test to what is returned now.

File: tests/test_api.py [hunk 2]
```diff
@@ -89,20 +89,6 @@ def test_domain_search():
...
-        assert 'status_codes' in response
-        assert 'suggestions' in response
-        assert 'query' in response
-        assert 'tlds' in response
-
-        for suggestion in api_call:
-            assert 'domain' in suggestion
-            assert 'status' in suggestion
-
-
```
Comment (ChuckWoodraska) [inline tests/test_api.py:105]: Took out the test because we took out the endpoint that is no longer available.

File: tests/test_cli.py [hunk 3]
```diff
@@ -15,8 +15,9 @@ def test_not_authenticated():
     (out_file, out_format, arguments) = cli.parse(args=['-c', 'non-existent', 'domain_search', 'google',
                                                         '--max-length', '100'])
     assert out_format == 'json'
-    assert not arguments.get('user', None)
-    assert not arguments.get('key', None)
+    print(arguments)
```
Comment (ChuckWoodraska) [inline tests/test_cli.py:18]: I will remove this in my next MR. I think we only get so many GitHub workflow actions and I would hate to waste it for an errant print statement in the tests.

File: tests/test_api.py [hunk 4]
```diff
@@ -296,12 +282,11 @@ def test_dict_like_behaviour():
         assert len(whois_google.items())
         assert len(whois_google.keys())
         assert len(whois_google.values())
-        assert whois_google.has_key('registrant')
```
Comment (ChuckWoodraska) [inline tests/test_api.py:299]: has_key is deprecated and this test does the same thing as the assert below.

**Review by kacieh80 — APPROVED** [review]
LGTM! Awesome job, I appreciate pointing out changes with the indention changes, thanks @ChuckWoodraska :shipit:

**Pull request merged**